src/Common/src/CoreLib/System/Memory.cs

```diff
@@ -19,7 +19,7 @@ namespace System
     /// </summary>
     [DebuggerTypeProxy(typeof(MemoryDebugView<>))]
     [DebuggerDisplay("{ToString(),raw}")]
-    public readonly struct Memory<T>
+    public readonly struct Memory<T> : IEquatable<Memory<T>>
```

Comment:
Changes to files in `CoreLib` need to made in the coreclr repo: https://github.com/dotnet/coreclr. Once its checked in, the changes will be mirrored here automatically. This PR (in corefx) should only update the `System.Runtime` reference assembly and add some tests.

1) Open up a PR in coreclr with the changes to the implementation:
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Memory.cs
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/ReadOnlyMemory.cs
2) Update this PR to change the type definition in the contract and add tests.
3) Wait for the mirror PR to merge.
4) Merge this change in with the PR which auto-updates the coreclr dependency.

This is what's causing the CI to fail:
```text
Build FAILED.

D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(72,5): error : CannotRemoveBaseTypeOrInterface : Type 'System.Memory<T>' does not implement interface 'System.IEquatable<System.Memory<T>>' in the implementation but it does in the contract. [D:\a\1\s\src\System.Runtime\src\System.Runtime.csproj]
D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(72,5): error : CannotRemoveBaseTypeOrInterface : Type 'System.ReadOnlyMemory<T>' does not implement interface 'System.IEquatable<System.ReadOnlyMemory<T>>' in the implementation but it does in the contract. [D:\a\1\s\src\System.Runtime\src\System.Runtime.csproj]
D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(85,5): error : ApiCompat failed for 'D:\a\1\s\artifacts\bin\System.Runtime\netcoreapp-Windows_NT-Debug\System.Runtime.dll' [D:\a\1\s\src\System.Runtime\src\System.Runtime.csproj]
D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(72,5): error : CannotRemoveBaseTypeOrInterface : Type 'System.Memory<T>' does not implement interface 'System.IEquatable<System.Memory<T>>' in the implementation but it does in the contract. [D:\a\1\s\src\System.Memory\src\System.Memory.csproj]
D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(72,5): error : CannotRemoveBaseTypeOrInterface : Type 'System.ReadOnlyMemory<T>' does not implement interface 'System.IEquatable<System.ReadOnlyMemory<T>>' in the implementation but it does in the contract. [D:\a\1\s\src\System.Memory\src\System.Memory.csproj]
D:\a\1\s\.packages\microsoft.dotnet.apicompat\1.0.0-beta.19121.4\build\Microsoft.DotNet.ApiCompat.targets(85,5): error : ApiCompat failed for 'D:\a\1\s\artifacts\bin\System.Memory\netcoreapp-Windows_NT-Debug\System.Memory.dll' [D:\a\1\s\src\System.Memory\src\System.Memory.csproj]
    0 Warning(s)
    6 Error(s)

```

Let me know if you have any questions around that. Thanks.